react/components/ModalDialog/index.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC, useEffect } from 'react'
+import PropTypes from 'prop-types'
+
+import Modal, { Props as ModalProps } from '../Modal'
+import Button from '../Button'
+
+type DialogOption = {
+  label: string
+  onClick: () => unknown
+}
+
+type Props = Pick<
+  ModalProps,
+  'children' | 'isOpen' | 'onClose' | 'onCloseTransitionFinish' | 'container'
+> & {
+  loading?: boolean
+  confirmation: DialogOption & { isDangerous?: boolean | null }
+  cancelation: DialogOption
+}
+
+const ModalDialog: FC<Props> = ({
+  children,
+  confirmation,
+  cancelation,
+  loading,
+  ...props
+}) => {
+  useEffect(() => {
+    console.warn(
+      'ModalDialog will be deprecated soon, please prefer to use Modal component. https://styleguide.vtex.com/#/Components/Overlays/Modal'
+    )
+  }, [])
+
+  return (
+    <Modal
+      {...props}
+      bottomBar={
+        <div>
+          <span className="mr4">
+            <Button
+              size="small"
+              type="button"
+              variation="tertiary"
+              disabled={loading}
+              onClick={() => {
+                cancelation.onClick?.()
+              }}
+            >
+              {cancelation.label}
+            </Button>
+          </span>
+          <Button
+            size="small"
+            type="button"
+            variation={confirmation.isDangerous ? 'danger' : 'primary'}
+            isLoading={loading}
+            onClick={() => {
+              confirmation.onClick?.()
+            }}
```

Comment:
```suggestion
            onClick={confirmation.onClick}
```

Reply:
The lint doesn't like this :rofl:

Reply:
:watt:

Reply:
Actually typescript doesn't like it, right? It shouldn't be the case, but ok then

Reply:
Actually it is the typescript. The `onClick` of the Button doesn't accept a function that returns unknown